app/messages/send-notification.js

```diff
@@ -0,0 +1,18 @@
+import { v4 as uuidv4 } from 'uuid'
+import { NotifyClient } from 'notifications-node-client'
+
+const sendNotification = async (message) => {
+  const notifyClient = new NotifyClient(process.env.NOTIFY_API_KEY)
+
+  await notifyClient.sendEmail(
+    message.body.data.notifyTemplateId,
+    message.body.data.commsAddress, {
+      personalisation: {
+        reference: message.body.data.personalisation.reference,
+        agreementSummaryLink: message.body.data.personalisation.agreementSummaryLink
+      },
```

Comment:
Could simplify this to

```
personalisation: message.body.personilisation
```

Then don't need to keep adapting for different templates.  Just use whatever personilsation provided.

Reply:
Have highlighted this to Amirs I. since I'm not sure how this changes the way he's structured the message payload within the work he's been doing on the message contract.

Reply:
Apologies for misunderstanding, have made the relevant update based on your suggestion:
```
      await notifyClient.sendEmail(
        message.body.data.notifyTemplateId,
        emailAddress, {
          personalisation: message.body.data.personalisation,
          reference: crypto.randomUUID()
        }

```